core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -7148,6 +7383,34 @@ private static class NonLiteralException extends RuntimeException {
         private static final NonLiteralException INSTANCE = new NonLiteralException();
     }
 
+    private class ChildTableColumnFinder implements PostOrderTreeTraversalAlgo.Visitor {
+        private boolean found;
+        private QueryModel targetModel;
+
+        public ChildTableColumnFinder(QueryModel targetModel, boolean found) {
+            this.targetModel = targetModel;
+            this.found = found;
+        }
+
+        @Override
+        public void visit(ExpressionNode node) throws SqlException {
+            if (node.type == LITERAL) {
+                CharSequence clauseColumn = node.token;
+                int dot = Chars.indexOfLastUnquoted(clauseColumn, '.');
+                String clauseColumnName = Chars.toString(clauseColumn, dot + 1, clauseColumn.length());
+                //check with column in between clause should be of master table
```

Comment:
What if a column is only present in the right-hand table and its literal has no table name prefix?